File: Create Binary Tree From Descriptions.py
# https://leetcode.com/problems/create-binary-tree-from-descriptions/?envType=daily-question&envId=2024-07-15
# Create Binary Tree From Descriptions

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def createBinaryTree(self, descriptions: List[List[int]]) -> Optional[TreeNode]:
        
        nodes = {}
        children = set()
        
        for parent, child, isLeft in descriptions:

            if parent not in nodes:
                nodes[parent] = TreeNode(val=parent)
            if child not in nodes:
                nodes[child] = TreeNode(val=child)
            
            if isLeft == 1:
                nodes[parent].left = nodes[child]
            else:
                nodes[parent].right = nodes[child]
            
            children.add(child)
        
        root = None

        for parent, _, _ in descriptions:

            if parent not in children:
                root = nodes[parent]
                break
        
        return root

File: Create Binary Tree From Descriptions.js
// https://leetcode.com/problems/create-binary-tree-from-descriptions/?envType=daily-question&envId=2026-06-07
// Create Binary Tree From Descriptions

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[][]} descriptions
 * @return {TreeNode}
 */
var createBinaryTree = function(descriptions) {
    
    const nodes = new Map();
    const children = new Set();

    for (const [parent, child, isLeft] of descriptions) {
        if (!nodes.has(parent)) {
            nodes.set(parent, new TreeNode(parent));
        }

        if (!nodes.has(child)) {
            nodes.set(child, new TreeNode(child));
        }

        if (isLeft === 1) {
            nodes.get(parent).left = nodes.get(child);
        } else {
            nodes.get(parent).right = nodes.get(child);
        }

        children.add(child);
    }

    let root = null;

    for (const [parent] of descriptions) {
        if (!children.has(parent)) {
            root = nodes.get(parent);
            break;
        }
    }

    return root;

};